content/post/051-new-ipfs-release-process.md

```diff
@@ -0,0 +1,229 @@
+    caught by testing the go-ipfs docker image in more production environments.
+  2. A CPU utilization regression in bitswap only seen under very high load. This
+    would have been caught by testing under production loads.
+  3. Panics in the DHT and QUIC modules that only show up under heavy load.
+
+We found two root causes:
+
+1. An uptick in development pace with respect to previous quarters without
+   matching improvements to our testing practices. Several large refactors
+   touching key but poorly tested subsystems landed in Q2.
+2. A significant increase in the network size and production demands on go-ipfs
+   without _any_ large-scale testing or network simulation infrastructure. In
+   the past all production scale testing has been done by deploying a custom
+   go-ipfs build to a bootstrapper or gateway and watching how it behaves.
+   
+To address these concerns, we've put a halt to all non-bugfix go-ipfs releases
+as we improve our testing practices and build out our testing and network
+simulation infrastructure. We felt this measure necessary as testing tends to
+become a secondary concern in the face new features and pressing performance
+issues.
+
+However, even with our current testing practices, these regressions should have
+been caught by pre-release testing. Furthermore, a better release process (e.g.,
+the ability to cut patch releases) would have enabled us to quickly release
+fixes for these regressions.
+
+Therefore, in addition to improving our testing, we're introducing a new release
+process to ensure that releases have been tested in as many environments as we
+can and that we can quickly release bug fixes without waiting an entire release
+cycle.
+
+## Release Process
+  
+We've made three specific changes to the release process:
+
+1. To address the stability issues, we've introduced a new release process that
+   involves extensively testing releases in a wide variety of production
+   environments.
+2. To address the issue of slow releases, we've introduced a 6 week release
+   cycle.
+3. To address the issue of slow bug fixes, we've switched to semver and
+   introduced patch releases. The first patch release will be 0.4.22 and the
+   next _feature_ release will be 0.5.0.
+   
+### Early Testers Program
+
+The early testers program allows groups using go-ipfs in production to
+self-volunteer to help with testing `go-ipfs` release candidates in development
+and production environments. While we invite the _entire_ community to help test
+releases, members of the early testers program participate directly and actively
+in every release.
+
+Early testers will deploy release candidates to dev and prod environments,
+giving us quick feedback on any regressions or performance changes they notice.
+This means we can get some quick feedback from heavy users before we cut the
+official release.
+
+This program currently includes:
+
+- Infura
+- Textile
+- Pinata
+- RTrade
+- QRI
+- Siderus
```

Comment:
We should at least link to their websites. People might read this and wonder what these awesome companies are doing with IPFS.

Reply:
Done!